src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/ExcelAdaptorServiceImpl.java

```diff
@@ -114,23 +114,43 @@ private void validateHeaders(List<String> headersToBeValidated,
         Field rowField = getRowIdField((Class<Object>) classType);
         Optional<Object> bean;
         while (rowIterator.hasNext()) {
-            bean = handleRowProcessing(classType, rowField, headers, rowIterator.next(), parentFieldMap,
-                childHeaderToCellMap, customObjectFieldsMapping);
-            bean.ifPresent(o -> objectList.add((T) o));
+            Row row = rowIterator.next();
+            if (!isBlankRow(row)) {
+                bean = handleRowProcessing(classType, rowField, headers, row, parentFieldMap,
+                        childHeaderToCellMap, customObjectFieldsMapping);
+                bean.ifPresent(o -> objectList.add((T) o));
+            } else {
+                break;
+            }
         }
         return objectList;
     }
 
+    private boolean isBlankRow(Row row) {
+        boolean isEmptyRow = false;
+        try {
+            if (checkIfRowIsEmpty(row)) {
+                if (row.getRowNum() == 1) {
```

Comment:
what about if rownum 1 blank and row 2 is valid...is it covered?

Reply:
@abhijitdiwan - I think it was decided if the first row is blank then we should not process the file itself

Reply:
@abhijitdiwan as Sujeeth mentioned, we have agreed that we do not process any records after a blank row, and in case of a file without any data in the first row we throw an exception, as it's considered as blank ([Observation 17](https://tools.hmcts.net/confluence/display/RTRD/Observations+-+CRD+Bugbash))